### PR TITLE
Serializer

### PR DESCRIPTION
**Pull request by dameert**
Serialization as a class responsability, removing the normalizer

### CODE REVIEW

File: Entity/SingleTypeIndex.php [hunk 1]
```diff
@@ -205,4 +207,20 @@ public function getContentType()
...
+     * @return mixed data which can be serialized by <b>json_encode</b>,
+     * which is a value of any type other than a resource.
+     * @since 5.4.0
+     */
+    public function jsonSerialize()
+    {
+        $json = new JsonClass(get_object_vars($this), __CLASS__);
+        $json->removeProperty('id');
+        $json->removeProperty('environment');
+
```
Comment (theus77) [inline Entity/SingleTypeIndex.php]: So, what the meaning of breaking this link? Is it not better to call the reconnect command?

Comment (dameert) [reply Entity/SingleTypeIndex.php]: When exporting and importing json files between projects, we cannot guarantee that the same environment names are used in both projects. This is also done for the contenttype json.

When importing, the environment to use should be specified (as we allready require for content types)

Comment (dameert) [reply Entity/SingleTypeIndex.php]: We cannot

> I'm not comfortable with the links between entities

Do you mean the "extends JsonDeserializable" ? Why are you not comfortable with that link?

**Comment by dameert** [comment]
> I'm not comfortable with the links between entities

Do you mean the "extends JsonDeserializable" ? Why are you not comfortable with that link?

**Pull request merged**